### PR TITLE
Authentication - JWT strategy - multiple audiences in config

### PR DESCRIPTION
**Pull request by AleF83**

### CODE REVIEW

File: services/src/modules/authentication/strategies/jwt.ts [hunk 1]
```diff
@@ -18,10 +19,15 @@ export default async function (request: fastify.FastifyRequest): Promise<void> {
...
+    const jwtAudiences = Array.isArray(decodedJWT.payload.aud)
       ? decodedJWT.payload.aud.map(a => String(a))
       : Array.of(String(decodedJWT.payload.aud));
-    if (!audience.includes(issuerConfig.audience)) {
+    const configAudiences = Array.isArray(issuerConfig.audience)
+      ? issuerConfig.audience
+      : Array.of(issuerConfig.audience);
+
+    const intersection = _.intersection(configAudiences, jwtAudiences);
+    if (!intersection || intersection.length === 0) {
```
Comment (thewizarodofoz) [inline services/src/modules/authentication/strategies/jwt.ts:30]: `_.intersection` will always return an array, we can drop `!intersection`

File: services/src/modules/authentication/strategies/jwt.ts [hunk 2]
```diff
@@ -18,10 +19,15 @@ export default async function (request: fastify.FastifyRequest): Promise<void> {
 
   // Verify audience
   if (issuerConfig.audience) {
-    const audience = Array.isArray(decodedJWT.payload.aud)
+    const jwtAudiences = Array.isArray(decodedJWT.payload.aud)
       ? decodedJWT.payload.aud.map(a => String(a))
       : Array.of(String(decodedJWT.payload.aud));
-    if (!audience.includes(issuerConfig.audience)) {
+    const configAudiences = Array.isArray(issuerConfig.audience)
```
Comment (thewizarodofoz) [inline services/src/modules/authentication/strategies/jwt.ts:25]: Why no something simpler like `[].concat(issuerConfig.audience)`?

File: services/tests/e2e/docker-compose.yml [hunk 3]
```diff
@@ -27,7 +27,7 @@ services:
             },
             "http://localhost:8070": {
               "authority": "http://authority-without-discrovery-endpoint",
-              "audience": "Stitch Gateway",
+              "audience": ["Stitch Gateway", "stitch gateway"],
```
Comment (thewizarodofoz) [inline services/tests/e2e/docker-compose.yml:30]: looks like now you are only testing multiple audience but not single audience

Comment (AleF83) [reply services/tests/e2e/docker-compose.yml:30]: There is another issues above on port 8060..

File: services/src/modules/authentication/strategies/jwt.ts (hunk 2 above)
Comment (AleF83) [reply services/src/modules/authentication/strategies/jwt.ts:25]: tried now.. `Type 'string' is not assignable to type 'ConcatArray<never>'`

Comment (AleF83) [reply services/src/modules/authentication/strategies/jwt.ts:25]: This works but it's ugly `([] as string[]).concat(issuerConfig.audience)`

**Review by thewizarodofoz — APPROVED** [review]

**Pull request merged**